projects/vdk-core/src/vdk/api/job_input.py

```diff
@@ -33,6 +33,26 @@ def set_all_properties(self, properties: dict):
         pass
 
 
+class ISecrets:
+    """
+    Allows for Data Job to store and retrieve sensitive data.
+    Secrets are solution for the following use cases
+    * Keeping API keys and passwords necessary to connect to different systems
+    """
+
+    @abstractmethod
+    def get_secret(self, name: str, default_value: Any = None) -> str:
```

Comment:
nit: name of what? I know it's also not documented in get_property but I was left wondering what is the expected format of name , just any string. 

Btw. what are the allow value formats. They are also not documented explicitly in IProperties but there are checks. Perhaps they should have been

It's a nit (feel free to ignore comment).

Reply:
will address them it in the upcoming MR